### PR TITLE
Task branch: filering added and tested manually; principal usage tested manually

### PR DESCRIPTION
**Pull request by Sofia-Ingl**

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|53.75% **`-8.48%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/b85da4fb550a1d99f0eab35c132ad6f9715232ba/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|88.59% **`-1.87%`**|:x:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/b85da4fb550a1d99f0eab35c132ad6f9715232ba/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|64.83% **`-14.83%`**|:x:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/b85da4fb550a1d99f0eab35c132ad6f9715232ba/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|48.57% **`-26.67%`**|:x:|
|[TaskSpecification.java](https://github.com/itmo-events-app/event-app/blob/b85da4fb550a1d99f0eab35c132ad6f9715232ba/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2Fspecification%2FTaskSpecification.java)|0%|:x:|

File: src/test/java/org/itmo/eventApp/main/controller/TaskControllerTest.java [hunk 1]
```diff
@@ -404,28 +404,28 @@ void taskMoveTest() throws Exception {
...
-//                        .contentType(MediaType.APPLICATION_JSON))
-//                .andExpect(status().isBadRequest());
-//
-//        task = taskRepository.findById(1).orElseThrow();
-//        Assertions.assertEquals(1, task.getEvent().getId());
-//
-//    }
+    @Test
+    void taskWrongMoveTest() throws Exception {
+        executeSqlScript("/sql/clean_tables.sql");
```
Comment (IP-13) [inline src/test/java/org/itmo/eventApp/main/controller/TaskControllerTest.java:409]: это же и тут перед каждым тестом исполняется

Comment (Sofia-Ingl) [reply src/test/java/org/itmo/eventApp/main/controller/TaskControllerTest.java:409]: Да? Не замечала. Уберу

File: src/main/java/org/itmo/eventapp/main/controller/TaskController.java [hunk 2]
```diff
@@ -150,27 +154,72 @@ public ResponseEntity<List<TaskResponse>> taskListCopy(
...
     @GetMapping("/event/{eventId}/where-assignee")
     public ResponseEntity<List<TaskResponse>> taskListShowInEventWhereAssignee(
             @Min(value = 1, message = "Параметр eventId не может быть меньше 1!")
-            @PathVariable Integer eventId
+            @PathVariable Integer eventId,
+            @RequestParam(required = false) Integer assignerId,
+            @RequestParam(required = false) TaskStatus taskStatus,
+            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime deadlineLowerLimit,
+            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime deadlineUpperLimit,
+            @RequestParam(required = false, defaultValue = "false") Boolean subEventTasksGet
```
Comment (Darklol) [inline src/main/java/org/itmo/eventapp/main/controller/TaskController.java:183]: Всё-таки правильнее для фильтра делать один @RequestParam filters
Можно MultiValueMap<String, String> делать, и проверять на существование соответствие ключа и значения в контроллере
В запросе передаём в виде /task?filters=key:value,key2:value2,key3:value3

**Review by IP-13 — APPROVED** [review]

**Comment by IP-13** [comment]
|Overall Project|53.8% **`-8.4%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/2a99c12b24cf6c404d2a844434ff376ba465e57e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|88.59% **`-1.87%`**|:x:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/2a99c12b24cf6c404d2a844434ff376ba465e57e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|64.83% **`-14.83%`**|:x:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/2a99c12b24cf6c404d2a844434ff376ba465e57e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|48.57% **`-26.67%`**|:x:|
|[TaskSpecification.java](https://github.com/itmo-events-app/event-app/blob/2a99c12b24cf6c404d2a844434ff376ba465e57e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2Fspecification%2FTaskSpecification.java)|0%|:x:|

**Pull request merged**